### PR TITLE
add 'bucket' parameter to storage-secret

### LINKED ISSUE #115: storage-config secret format is different between kserve and modelmesh
odh-model-controller is responsible for generating `storage-config` which is the main secret to accessing models.

For modelmesh, the secret `storage-config` is 
~~~
{"access_key_id":"AKIAIOSFODNN7EXAMPLE","default_bucket":"modelmesh-example-models","endpoint_url":"http://minio:9000","region":"us-south","secret_access_key":"wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY","type":"s3"}
~~~

but for kserve, it looks like
~~~
apiVersion: v1
data:
  AWS_ACCESS_KEY_ID: YWRtaW4=
  AWS_SECRET_ACCESS_KEY: cGFzc3dvcmQ=
kind: Secret
metadata:
  annotations:
    serving.kserve.io/s3-endpoint: minio.minio.svc:9000
    serving.kserve.io/s3-region: us-east-2
    serving.kserve.io/s3-useanoncredential: "false"
    serving.kserve.io/s3-usehttps: "0"
  name: storage-config
~~~

odh-model-controller needs to generate right format of storage-config secret based on deployment type(kserve, modelmesh)

### PR DESCRIPTION
**Pull request by VedantMahabaleshwarkar**
## Description

Fixes #115 
Add `bucket` parameter to the storage-secret created by odh-model-controller. This is in addition to the already existing `default_bucket` parameter in the storage-secret. 
This is because kserve and modelmesh have slightly different formats for the secret. Kserve expects the bucket to be specified in  `bucket`, but ModelMesh expects the bucket to be specified in `default_bucket`. Long term solution should be to create an upstream issue to have the formats match upstream. For now, adding both keys as parameter having the same value should make the secret work for both kserve and modelmesh. 

## How Has This Been Tested?

- Local changes were built and pushed to `quay.io/vedantm/odh-model-controller:storage_secret_fix` 
- Created a branch `storage_secret_fix_test` which modifies `params.env` to deploy `quay.io/vedantm/odh-model-controller:storage_secret_fix`  
- Tested changes using the custom image and branch. 

### Testing instructions : 
- Install Kserve dependencies 
- Install ODH operator 2.3+ 
- Create the following DSC 
```
cat <<EOF |oc create -f - 
spec:
  components:
    dashboard:
      devFlags:
        manifests:
          - contextDir: manifests
            sourcePath: ''
            uri: >-
              https://github.com/opendatahub-io/odh-dashboard/tarball/f/model-serving
      managementState: Managed
    kserve:
      devFlags:
        manifests:
          - contextDir: config
            sourcePath: overlays/odh
            uri: 'https://github.com/opendatahub-io/kserve/tarball/release-v0.11.1'
          - contextDir: config
            sourcePath: ''
            uri: >-
              https://github.com/VedantMahabaleshwarkar/odh-model-controller/tarball/storage_secret_fix_test
      managementState: Managed
EOF
``` 
- Log in to ODH Dashboard
- Create a Data Science Project `test` 
- Create Data Connection called `rhodss3` for the `rhods-public` s3 bucket (it's a public bucket so your own aws access+secret key should work, region is us-east-1)
- Create ServingRuntime 
```
cat <<EOF |oc create -f - 
apiVersion: serving.kserve.io/v1alpha1
kind: ServingRuntime
metadata:
  name: caikit-runtime
spec:
  containers:
  - env:
    - name: RUNTIME_LOCAL_MODELS_DIR
      value: /mnt/models
    # Last tested image stable-27c0d61
    image: quay.io/opendatahub/caikit-tgis-serving:stable   
    name: kserve-container
    ports:
    # Note, KServe only allows a single port, this is the gRPC port. Subject to change in the future
    - containerPort: 8085
      name: h2c
      protocol: TCP
    resources:
      requests:
        cpu: 4
        memory: 8Gi
  multiModel: false
  supportedModelFormats:
  # Note: this currently *only* supports caikit format models
  - autoSelect: true
    name: caikit
EOF
```
- Create InferenceService 
```
cat <<EOF |oc create -f - 
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  annotations:
    serving.knative.openshift.io/enablePassthrough: "true"
    sidecar.istio.io/inject: "true"
    sidecar.istio.io/rewriteAppHTTPProbers: "true"
  name: caikit-example-isvc
spec:
  predictor:
    model:
      modelFormat:
        name: caikit
      runtime: caikit-runtime
      storage:
        key: aws-connection-rhodss3
        path: demo-models/flan-t5-small-caikit/flan-t5-small-caikit/
EOF
```
- Check isvc successfully deploys 

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by Jooho** [comment]
`make test` change config/rbac/role.yaml because of linter. could you please execute the command before sending a pr?
It is just for formating stuff.

**Comment by VedantMahabaleshwarkar** [comment]
@Jooho added the linter changes to `role.yaml` as well

File: controllers/storageconfig_controller.go [hunk 1]
```diff
@@ -53,7 +54,10 @@ func newStorageSecret(dataConnectionSecretsList *corev1.SecretList) *corev1.Secr
 		dataConnectionElement["access_key_id"] = string(secret.Data["AWS_ACCESS_KEY_ID"])
 		dataConnectionElement["secret_access_key"] = string(secret.Data["AWS_SECRET_ACCESS_KEY"])
 		dataConnectionElement["endpoint_url"] = string(secret.Data["AWS_S3_ENDPOINT"])
+		// We add both "default_bucket" and "bucket" because kserve and modelmesh expect different keys for the same value
+		// Once upstream has reconciled to one common key, we should remove the other one.
 		dataConnectionElement["default_bucket"] = string(secret.Data["AWS_S3_BUCKET"])
+		dataConnectionElement["bucket"] = string(secret.Data["AWS_S3_BUCKET"])
```
Comment (spolti) [inline controllers/storageconfig_controller.go:60]: Is there any unit test for this?

**Comment by christianvogt** [comment]
I tested this scenario and the change works as intended. I'm encountering other issues with the service route not coming up but that seems to be separate.

**Comment by lucferbux** [comment]
> I tested this scenario and the change works as intended. I'm encountering other issues with the service route not coming up but that seems to be separate.

Ok, I've replicated the issue and this seems the solution, thanks a lot @VedantMahabaleshwarkar

File: controllers/storageconfig_controller.go (hunk 1 above)
Comment (VedantMahabaleshwarkar) [reply controllers/storageconfig_controller.go:60]: covered by the changes to `controllers/testdata/secrets/storageconfig-encoded.yaml` where the new encoded value expects the new `bucket` key

**Review by Jooho — APPROVED** [review]
/lgtm

**Pull request merged**